frontend/src/metabase-lib/expressions/resolver.js

```diff
@@ -69,12 +69,19 @@ const isCompatible = (a, b) => {
 };
 
 /**
- *
- * @param {import("./pratt").Expr} expression
- * @param {string} type
- * @param {?(kind: string, name: string, node: import("./pratt").Node) => void} fn
+ * @param {{
+ *   expression: import("./pratt").Expr
+ *   type?: string
+ *   fn?: ?(kind: string, name: string, node: import("./pratt").Node) => void
+ *   database?: Database | null
+ * }} options
  */
-export function resolve(expression, type = "expression", fn = undefined) {
+export function resolve({
+  expression,
+  type = "expression",
+  fn = undefined,
+  database = undefined,
```

Comment:
Should we make `database` required and pass it in all of the calls?
e2e tests for custom columns are failing, this might be related.

Reply:
Not sure what to do here: the tests are broken because they incorrectly assumed that the built in database supports the `convert-timezone` features, whereas it does not.

I think the only correct way to fix this is to remove that case from the test, but then we don't have a test for those functions anymore.